metadata.yaml

```diff
@@ -37,6 +37,9 @@ requires:
     interface: tls-certificates
     limit: 1
     optional: true
+  s3-credentials:
+    interface: s3
+    limit: 1
```

Comment:
**nitpick/todo:** I think we should add `optional: true` here. There's some work in the pipeline I believe for Charmcraft to auto-generate stuff based on `metadata.yaml`, so would be good here.

Reply:
Fixed in 40ec675849db9e0df2fcf0ada2953a41d873418d